### PR TITLE
Fix applying target column options on foreign key columns

### PR DESCRIPTION
**Pull request by mrxotey**
Fixes doctrine/dbal#2811
Similar PR: #6830 

Tobion's PR fixes only applying of collation, which is not enough. This fix applies all options from target column to make type of source and target columns equal.

P.S.
Sorry for my terrible english.

### CODE REVIEW

File: lib/Doctrine/ORM/Tools/SchemaTool.php [hunk 1]
```diff
@@ -871,4 +857,21 @@ public function getUpdateSchemaSql(array $classes, $saveMode = false)
 
         return $schemaDiff->toSql($this->platform);
     }
+
+    private function gatherColumnOptions(array &$options, array $mapping)
+    {
+        if (isset($mapping['options'])) {
+            $knownOptions = array('comment', 'unsigned', 'fixed', 'default');
```
Comment (Tobion) [inline lib/Doctrine/ORM/Tools/SchemaTool.php]: I don't think it makes sense to copy the column comment from the PK to the foreign key. So that should probably be excluded, but makes it kinda more ugly.

File: tests/Doctrine/Tests/Mocks/HydratorMockStatement.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function __construct(array $resultSet)
      *
      * @return array
      */
-    public function fetchAll($fetchStyle = null, $columnIndex = null, array $ctorArgs = null)
+    public function fetchAll($fetchStyle = null, $columnIndex = null, $ctorArgs = null)
```
Comment (Tobion) [inline tests/Doctrine/Tests/Mocks/HydratorMockStatement.php]: why was all those unrelated stuff changed?

File: tests/Doctrine/Tests/Models/Forum/ForumCategory.php [hunk 3]
```diff
@@ -9,7 +9,7 @@
 class ForumCategory
 {
     /**
-     * @Column(type="integer")
+     * @Column(type="guid", options={"fixed":true, "collation":"latin1_bin", "foo":"bar"})
```
Comment (Tobion) [inline tests/Doctrine/Tests/Models/Forum/ForumCategory.php]: guid might not be a string so applying fixed or collation might not make sense in this case. maybe just use a string simply.

**Comment by Tobion** [comment]
Fixing it generically makes sense 👍

**Comment by Tobion** [comment]
@mrxotey ping

**Comment by lcobucci** [comment]
@mrxotey @Tobion as mentioned on our [contributions guidelines](https://github.com/doctrine/doctrine2/blob/master/CONTRIBUTING.md#we-only-accept-prs--to-master) we accept PRs only to `master` and we handle the backporting if needed (and when possible). With that said, please let's apply the fixes on #6830.

**Comment by lcobucci** [comment]
I'll close this PR as `invalid` as per my comment.

**Comment by mrxotey** [comment]
Sorry. Was busy. I'll create new PR with review fixes. Thanks.
